src/main/java/com/getindata/connectors/http/SinkHttpClient.java

```diff
@@ -0,0 +1,22 @@
+package com.getindata.connectors.http;
+
+import com.getindata.connectors.http.sink.HttpSinkRequestEntry;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * An HTTP client that is used by {@link com.getindata.connectors.http.sink.HttpSinkWriter}
+ * to send HTTP requests processed by {@link com.getindata.connectors.http.sink.HttpSink}.
+ */
+public interface SinkHttpClient {
+  /**
+   * Returns a {@link CompletableFuture} that completes when all requests have been sent and returned their statuses.
```

Comment:
You can place this line under

@returns tag, after parameters.

As a general description you can write somethign like "sends REST request to external web service" or something like that.